packages/contracts/src/contracts/current/protocol/AssetTransferProxy/AssetProxyEncoderDecoder.sol

```diff
@@ -0,0 +1,98 @@
+/*
+
+  Copyright 2018 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.4.21;
+
+import "../../utils/Memory/Memory.sol";
+
+contract AssetProxyEncoderDecoder is
+    Memory
+{
+    // Supported asset proxies
+    enum AssetProxyId {
+        ERC20,
+        ERC721,
+        ENS,
+        OWNABLE,
+        ACCEPT_OWNERSHIP
```

Comment:
Yeah I think that's fine.

I put the full list in so that we wouldn't have to redeploy the AssetTransferProxy contract each time we add one of these proxies. Although after chatting with Amir, I think we're going to change AssetTransferProxy's API to not depend on this enum at all.